src/components/spinner/index.tsx

```diff
@@ -87,7 +87,8 @@ const Spinner = () => {
   // prevents spinner from showing when loading is < 100ms
   // makes user experience less jarring
   useEffect(() => {
-    setTimeout(() => setIsShown(true), 100)
+    const t = setTimeout(() => setIsShown(true), 100)
+    return () => clearTimeout(t)
```

Comment:
Here was the bug. It is a minor bug. The timeout was sometimes firing after the component unmounted, because I didn't have a cleanup to clear the timeout.